configure.ac

```diff
@@ -56,6 +56,28 @@ AM_GNU_GETTEXT_VERSION([0.18.2])
 AM_GNU_GETTEXT([external])
 AM_ICONV_LINK
 
+AM_CFLAGS="\
+ -Wall\
+ -Wdeclaration-after-statement\
+ -Wextra\
+ -Wmissing-format-attribute\
+ -Wmissing-noreturn\
+ -Wno-gnu-alignof-expression\
+ -Wpointer-arith\
+ -Wshadow\
+ -Wstrict-prototypes\
+ -Wundef\
+ -Wunused\
+ -Wwrite-strings"
+
```

Comment:
We can't assume gcc or clang as the compiler, basically any special flags beyond -Wall (which appears to be rather universal) would need to be tested on whether the current compiler supports it.